pkg/storageincentives/agent.go

```diff
@@ -268,21 +281,30 @@ func (a *Agent) start(blockTime time.Duration, blocksPerRound, blocksPerPhase ui
 		}
 
 		prevPhase = currentPhase
-
 		mtx.Unlock()
+		a.mtx.Lock()
+		a.nodeStatus.Block = block
+		a.nodeStatus.State = currentPhase.String()
+		a.nodeStatus.Round = round
+		a.mtx.Unlock()
+		a.saveStatus()
 	}
 }
 
 func (a *Agent) reveal(ctx context.Context, storageRadius uint8, sample, obfuscationKey []byte) error {
 	a.metrics.RevealPhase.Inc()
 	err := a.contract.Reveal(ctx, storageRadius, sample, obfuscationKey)
+	a.setNodeStatusFee(a.contract.GetFee())
 	if err != nil {
 		a.metrics.ErrReveal.Inc()
 	}
 	return err
 }
 
 func (a *Agent) claim(ctx context.Context) error {
+	defer func() {
+		a.saveStatus()
```

Comment:
I think the code can be cleaned up by moving the new "state" related things to it's own separate package under the `storageincentives`.